### ISSUE TITLE
Panic in light client

### ISSUE THREAD

**Issue by viquezclaudio** [nimiq/core-rs-albatross#1264]
Panic in the light client while connected to the public devnet 

```
2023-01-17T20:48:36.945634Z DEBUG sync_requests        | Received epoch_ids received_epochs=0 start_epoch=3 checkpoint=true sender=12D3KooWLJkYiTkTnkcC9xq4Q5MnNrQeLbsgYdwASoUjrCBm9eHB
2023-01-17T20:48:37.297425Z DEBUG sync_requests        | Received epoch_ids received_epochs=0 start_epoch=3 checkpoint=true sender=12D3KooWCvMyCS91oDSNkFHKapGPRFm1a9vkz72eVe15ERoDb2r1
2023-01-17T20:48:37.303664Z DEBUG sync_requests        | Received epoch_ids received_epochs=0 start_epoch=3 checkpoint=true sender=12D3KooWCvMyCS91oDSNkFHKapGPRFm1a9vkz72eVe15ERoDb2r1
2023-01-17T20:48:37.663343Z DEBUG sync_stream          | Pushed a macro block block_number=60600 push_result=Extended
2023-01-17T20:48:37.663700Z DEBUG sync_stream          | Pushed a macro block block_number=60600 push_result=Known
2023-01-17T20:48:38.800607Z DEBUG behaviour            | Doing house-keeping in peer address book
2023-01-17T20:48:38.800863Z DEBUG peer_contacts        | Adding peer contact peer_id=12D3KooWFMJrQcKiE97krgRf2c1U5DdEWs2w8kDgRCkbWPi1UXCP
2023-01-17T20:48:38.801791Z DEBUG behaviour            | Maintaining peers peer_ids=connected=11, dialing=0, failed=0, down=0 addresses=connected=11, dialing=0, failed=25, down=0
2023-01-17T20:48:38.852894Z INFO  nimiq_light_client   | Consensus status: true - Head: #60600- b2419d4d2b26b8766cfc9abbe889a03dcd135d1913663500657c98cbcb03a291 block_number=60600 num_peers=10 status=true
2023-01-17T20:48:38.872977Z ERROR panic                | thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()` on a `None` value': /Users/claudioviquez/Workspace/core-rs-albatross/consensus/src/sync/light/sync_stream.rs:172
   0: backtrace::backtrace::libunwind::trace
             at /Users/claudioviquez/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.64/src/backtrace/libunwind.rs:93:5
      backtrace::backtrace::trace_unsynchronized
             at /Users/claudioviquez/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.64/src/backtrace/mod.rs:66:5
   1: backtrace::backtrace::trace
             at /Users/claudioviquez/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.64/src/backtrace/mod.rs:53:14
   2: backtrace::capture::Backtrace::create
             at /Users/claudioviquez/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.64/src/capture.rs:176:9
   3: backtrace::capture::Backtrace::new
             at /Users/claudioviquez/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.64/src/capture.rs:140:22
   4: <backtrace::capture::Backtrace as core::default::Default>::default
             at /Users/claudioviquez/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.64/src/capture.rs:410:9
   5: log_panics::Config::install_panic_hook::{{closure}}
             at /Users/claudioviquez/.cargo/registry/src/github.com-1ecc6299db9ec823/log-panics-2.1.0/src/lib.rs:115:29
   6: std::panicking::rust_panic_with_hook
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:702:17
   7: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:586:13
   8: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/sys_common/backtrace.rs:138:18
   9: rust_begin_unwind
```

**Comment by jsdanielh** [comment]
Also seen in the web-client:
```
2023-02-08T16:37:58.158000128Z ERROR panic                | thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value': consensus/src/sync/light/sync_stream.rs:191    

imports.wbg.__wbg_error_ef9a0be47931175f @ nimiq_web_client.js:515
$func5085 @ nimiq_web_client_bg.wasm:0x405fa9
$func999 @ nimiq_web_client_bg.wasm:0x296694
$func4174 @ nimiq_web_client_bg.wasm:0x3f0deb
$func1449 @ nimiq_web_client_bg.wasm:0x2f1b76
$func2130 @ nimiq_web_client_bg.wasm:0x351b53
$func3986 @ nimiq_web_client_bg.wasm:0x3e8385
$func920 @ nimiq_web_client_bg.wasm:0x282b2e
$func3918 @ nimiq_web_client_bg.wasm:0x3e4ca3
$func4234 @ nimiq_web_client_bg.wasm:0x3f33ad
$func4954 @ nimiq_web_client_bg.wasm:0x404b00
$func4840 @ nimiq_web_client_bg.wasm:0x403304
$func4859 @ nimiq_web_client_bg.wasm:0x40379d
$func4708 @ nimiq_web_client_bg.wasm:0x400d4d
$func96 @ nimiq_web_client_bg.wasm:0x71dae
$func136 @ nimiq_web_client_bg.wasm:0xe0131
$func168 @ nimiq_web_client_bg.wasm:0x113fdb
$func1794 @ nimiq_web_client_bg.wasm:0x3269f5
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h625215fdc896e2a6 @ nimiq_web_client_bg.wasm:0x40600d
__wbg_adapter_40 @ nimiq_web_client.js:220
real @ nimiq_web_client.js:189
Promise.then (async)
imports.wbg.__wbg_then_11f7a54d67b4bfad @ nimiq_web_client.js:431
$func3598 @ nimiq_web_client_bg.wasm:0x3d2d48
$func3974 @ nimiq_web_client_bg.wasm:0x3e799a
$func3412 @ nimiq_web_client_bg.wasm:0x3c7067
$func1607 @ nimiq_web_client_bg.wasm:0x30b822
$func76 @ nimiq_web_client_bg.wasm:0xd859
$func99 @ nimiq_web_client_bg.wasm:0x7eb49
$func708 @ nimiq_web_client_bg.wasm:0x24458a
$func5158 @ nimiq_web_client_bg.wasm:0x40674e
$func1794 @ nimiq_web_client_bg.wasm:0x3269f5
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h625215fdc896e2a6 @ nimiq_web_client_bg.wasm:0x40600d
__wbg_adapter_40 @ nimiq_web_client.js:220
real @ nimiq_web_client.js:189
Promise.then (async)
imports.wbg.__wbg_then_11f7a54d67b4bfad @ nimiq_web_client.js:431
$func3598 @ nimiq_web_client_bg.wasm:0x3d2d48
$func3974 @ nimiq_web_client_bg.wasm:0x3e799a
$func1026 @ nimiq_web_client_bg.wasm:0x29cde6
$func3560 @ nimiq_web_client_bg.wasm:0x3d07d5
$func365 @ nimiq_web_client_bg.wasm:0x1b6ae0
$func5158 @ nimiq_web_client_bg.wasm:0x40674e
$func1794 @ nimiq_web_client_bg.wasm:0x3269f5
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h625215fdc896e2a6 @ nimiq_web_client_bg.wasm:0x40600d
__wbg_adapter_40 @ nimiq_web_client.js:220
real @ nimiq_web_client.js:189
Promise.then (async)
imports.wbg.__wbg_then_11f7a54d67b4bfad @ nimiq_web_client.js:431
$func3598 @ nimiq_web_client_bg.wasm:0x3d2d48
$func3974 @ nimiq_web_client_bg.wasm:0x3e799a
$func3001 @ nimiq_web_client_bg.wasm:0x3a88d5
$func5409 @ nimiq_web_client_bg.wasm:0x4077da
$func188 @ nimiq_web_client_bg.wasm:0x12d814
$func198 @ nimiq_web_client_bg.wasm:0x138896
$func113 @ nimiq_web_client_bg.wasm:0xaa9a4
$func1199 @ nimiq_web_client_bg.wasm:0x2c28ad
$func3269 @ nimiq_web_client_bg.wasm:0x3bd3bf
$func917 @ nimiq_web_client_bg.wasm:0x281c41
$func3913 @ nimiq_web_client_bg.wasm:0x3e47ec
$func204 @ nimiq_web_client_bg.wasm:0x13f479
$func365 @ nimiq_web_client_bg.wasm:0x1b6b56
$func5158 @ nimiq_web_client_bg.wasm:0x40674e
$func1794 @ nimiq_web_client_bg.wasm:0x3269f5
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h625215fdc896e2a6 @ nimiq_web_client_bg.wasm:0x40600d
__wbg_adapter_40 @ nimiq_web_client.js:220
real @ nimiq_web_client.js:189
Promise.then (async)
imports.wbg.__wbg_then_11f7a54d67b4bfad @ nimiq_web_client.js:431
$func3598 @ nimiq_web_client_bg.wasm:0x3d2d48
$func4678 @ nimiq_web_client_bg.wasm:0x40037d
$func2728 @ nimiq_web_client_bg.wasm:0x3901e2
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h0e72e882800baf25 @ nimiq_web_client_bg.wasm:0x405e96
__wbg_adapter_27 @ nimiq_web_client.js:208
real @ nimiq_web_client.js:189
nimiq_web_client_bg.wasm:0x3e4cc6 Uncaught (in promise) RuntimeError: unreachable
    at nimiq_web_client_bg.wasm:0x3e4cc6
    at nimiq_web_client_bg.wasm:0x3f33ad
    at nimiq_web_client_bg.wasm:0x404b00
    at nimiq_web_client_bg.wasm:0x403304
    at nimiq_web_client_bg.wasm:0x40379d
    at nimiq_web_client_bg.wasm:0x400d4d
    at nimiq_web_client_bg.wasm:0x71dae
    at nimiq_web_client_bg.wasm:0xe0131
    at nimiq_web_client_bg.wasm:0x113fdb
    at nimiq_web_client_bg.wasm:0x3269f5
$func3918 @ nimiq_web_client_bg.wasm:0x3e4cc6
$func4234 @ nimiq_web_client_bg.wasm:0x3f33ad
$func4954 @ nimiq_web_client_bg.wasm:0x404b00
$func4840 @ nimiq_web_client_bg.wasm:0x403304
$func4859 @ nimiq_web_client_bg.wasm:0x40379d
$func4708 @ nimiq_web_client_bg.wasm:0x400d4d
$func96 @ nimiq_web_client_bg.wasm:0x71dae
$func136 @ nimiq_web_client_bg.wasm:0xe0131
$func168 @ nimiq_web_client_bg.wasm:0x113fdb
$func1794 @ nimiq_web_client_bg.wasm:0x3269f5
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h625215fdc896e2a6 @ nimiq_web_client_bg.wasm:0x40600d
__wbg_adapter_40 @ nimiq_web_client.js:220
real @ nimiq_web_client.js:189
Promise.then (async)
imports.wbg.__wbg_then_11f7a54d67b4bfad @ nimiq_web_client.js:431
$func3598 @ nimiq_web_client_bg.wasm:0x3d2d48
$func3974 @ nimiq_web_client_bg.wasm:0x3e799a
$func3412 @ nimiq_web_client_bg.wasm:0x3c7067
$func1607 @ nimiq_web_client_bg.wasm:0x30b822
$func76 @ nimiq_web_client_bg.wasm:0xd859
$func99 @ nimiq_web_client_bg.wasm:0x7eb49
$func708 @ nimiq_web_client_bg.wasm:0x24458a
$func5158 @ nimiq_web_client_bg.wasm:0x40674e
$func1794 @ nimiq_web_client_bg.wasm:0x3269f5
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h625215fdc896e2a6 @ nimiq_web_client_bg.wasm:0x40600d
__wbg_adapter_40 @ nimiq_web_client.js:220
real @ nimiq_web_client.js:189
Promise.then (async)
imports.wbg.__wbg_then_11f7a54d67b4bfad @ nimiq_web_client.js:431
$func3598 @ nimiq_web_client_bg.wasm:0x3d2d48
$func3974 @ nimiq_web_client_bg.wasm:0x3e799a
$func1026 @ nimiq_web_client_bg.wasm:0x29cde6
$func3560 @ nimiq_web_client_bg.wasm:0x3d07d5
$func365 @ nimiq_web_client_bg.wasm:0x1b6ae0
$func5158 @ nimiq_web_client_bg.wasm:0x40674e
$func1794 @ nimiq_web_client_bg.wasm:0x3269f5
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h625215fdc896e2a6 @ nimiq_web_client_bg.wasm:0x40600d
__wbg_adapter_40 @ nimiq_web_client.js:220
real @ nimiq_web_client.js:189
Promise.then (async)
imports.wbg.__wbg_then_11f7a54d67b4bfad @ nimiq_web_client.js:431
$func3598 @ nimiq_web_client_bg.wasm:0x3d2d48
$func3974 @ nimiq_web_client_bg.wasm:0x3e799a
$func3001 @ nimiq_web_client_bg.wasm:0x3a88d5
$func5409 @ nimiq_web_client_bg.wasm:0x4077da
$func188 @ nimiq_web_client_bg.wasm:0x12d814
$func198 @ nimiq_web_client_bg.wasm:0x138896
$func113 @ nimiq_web_client_bg.wasm:0xaa9a4
$func1199 @ nimiq_web_client_bg.wasm:0x2c28ad
$func3269 @ nimiq_web_client_bg.wasm:0x3bd3bf
$func917 @ nimiq_web_client_bg.wasm:0x281c41
$func3913 @ nimiq_web_client_bg.wasm:0x3e47ec
$func204 @ nimiq_web_client_bg.wasm:0x13f479
$func365 @ nimiq_web_client_bg.wasm:0x1b6b56
$func5158 @ nimiq_web_client_bg.wasm:0x40674e
$func1794 @ nimiq_web_client_bg.wasm:0x3269f5
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h625215fdc896e2a6 @ nimiq_web_client_bg.wasm:0x40600d
__wbg_adapter_40 @ nimiq_web_client.js:220
real @ nimiq_web_client.js:189
Promise.then (async)
imports.wbg.__wbg_then_11f7a54d67b4bfad @ nimiq_web_client.js:431
$func3598 @ nimiq_web_client_bg.wasm:0x3d2d48
$func4678 @ nimiq_web_client_bg.wasm:0x40037d
$func2728 @ nimiq_web_client_bg.wasm:0x3901e2
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h0e72e882800baf25 @ nimiq_web_client_bg.wasm:0x405e96
__wbg_adapter_27 @ nimiq_web_client.js:208
real @ nimiq_web_client.js:189
nimiq_web_client.js:515 2023-02-08T16:37:58.360999936Z ERROR panic                | thread '<unnamed>' panicked at 'already borrowed: BorrowMutError': /home/jsdaniel/.cargo/registry/src/github.com-1ecc6299db9ec823/wasm-bindgen-futures-0.4.33/src/task/singlethread.rs:85    

imports.wbg.__wbg_error_ef9a0be47931175f @ nimiq_web_client.js:515
$func5085 @ nimiq_web_client_bg.wasm:0x405fa9
$func999 @ nimiq_web_client_bg.wasm:0x296694
$func4174 @ nimiq_web_client_bg.wasm:0x3f0deb
$func1449 @ nimiq_web_client_bg.wasm:0x2f1b76
$func2130 @ nimiq_web_client_bg.wasm:0x351b53
$func3986 @ nimiq_web_client_bg.wasm:0x3e8385
$func920 @ nimiq_web_client_bg.wasm:0x282b2e
$func3918 @ nimiq_web_client_bg.wasm:0x3e4ca3
$func4234 @ nimiq_web_client_bg.wasm:0x3f33db
$func4954 @ nimiq_web_client_bg.wasm:0x404b00
$func4840 @ nimiq_web_client_bg.wasm:0x403304
$func4859 @ nimiq_web_client_bg.wasm:0x40379d
$func4306 @ nimiq_web_client_bg.wasm:0x3f5eb8
$func1794 @ nimiq_web_client_bg.wasm:0x326b3c
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h625215fdc896e2a6 @ nimiq_web_client_bg.wasm:0x40600d
__wbg_adapter_40 @ nimiq_web_client.js:220
real @ nimiq_web_client.js:189
Promise.then (async)
imports.wbg.__wbg_then_11f7a54d67b4bfad @ nimiq_web_client.js:431
$func3598 @ nimiq_web_client_bg.wasm:0x3d2d48
$func3974 @ nimiq_web_client_bg.wasm:0x3e799a
$func3412 @ nimiq_web_client_bg.wasm:0x3c7067
$func1607 @ nimiq_web_client_bg.wasm:0x30b822
$func76 @ nimiq_web_client_bg.wasm:0xd859
$func99 @ nimiq_web_client_bg.wasm:0x7eb49
$func708 @ nimiq_web_client_bg.wasm:0x24458a
$func5158 @ nimiq_web_client_bg.wasm:0x40674e
$func1794 @ nimiq_web_client_bg.wasm:0x3269f5
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h625215fdc896e2a6 @ nimiq_web_client_bg.wasm:0x40600d
__wbg_adapter_40 @ nimiq_web_client.js:220
real @ nimiq_web_client.js:189
Promise.then (async)
imports.wbg.__wbg_then_11f7a54d67b4bfad @ nimiq_web_client.js:431
$func3598 @ nimiq_web_client_bg.wasm:0x3d2d48
$func3974 @ nimiq_web_client_bg.wasm:0x3e799a
$func1026 @ nimiq_web_client_bg.wasm:0x29cde6
$func3560 @ nimiq_web_client_bg.wasm:0x3d07d5
$func365 @ nimiq_web_client_bg.wasm:0x1b6ae0
$func5158 @ nimiq_web_client_bg.wasm:0x40674e
$func1794 @ nimiq_web_client_bg.wasm:0x3269f5
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h625215fdc896e2a6 @ nimiq_web_client_bg.wasm:0x40600d
__wbg_adapter_40 @ nimiq_web_client.js:220
real @ nimiq_web_client.js:189
Promise.then (async)
imports.wbg.__wbg_then_11f7a54d67b4bfad @ nimiq_web_client.js:431
$func3598 @ nimiq_web_client_bg.wasm:0x3d2d48
$func3974 @ nimiq_web_client_bg.wasm:0x3e799a
$func3001 @ nimiq_web_client_bg.wasm:0x3a88d5
$func5409 @ nimiq_web_client_bg.wasm:0x4077da
$func188 @ nimiq_web_client_bg.wasm:0x12d814
$func198 @ nimiq_web_client_bg.wasm:0x138896
$func113 @ nimiq_web_client_bg.wasm:0xaa9a4
$func1199 @ nimiq_web_client_bg.wasm:0x2c28ad
$func3269 @ nimiq_web_client_bg.wasm:0x3bd3bf
$func917 @ nimiq_web_client_bg.wasm:0x281c41
$func3913 @ nimiq_web_client_bg.wasm:0x3e47ec
$func204 @ nimiq_web_client_bg.wasm:0x13f479
$func365 @ nimiq_web_client_bg.wasm:0x1b6b56
$func5158 @ nimiq_web_client_bg.wasm:0x40674e
$func1794 @ nimiq_web_client_bg.wasm:0x3269f5
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h625215fdc896e2a6 @ nimiq_web_client_bg.wasm:0x40600d
__wbg_adapter_40 @ nimiq_web_client.js:220
real @ nimiq_web_client.js:189
Promise.then (async)
imports.wbg.__wbg_then_11f7a54d67b4bfad @ nimiq_web_client.js:431
$func3598 @ nimiq_web_client_bg.wasm:0x3d2d48
$func4678 @ nimiq_web_client_bg.wasm:0x40037d
$func2728 @ nimiq_web_client_bg.wasm:0x3901e2
$_dyn_core__ops__function__FnMut__A____Output___R_as_wasm_bindgen__closure__WasmClosure___describe__invoke__h0e72e882800baf25 @ nimiq_web_client_bg.wasm:0x405e96
__wbg_adapter_27 @ nimiq_web_client.js:208
real @ nimiq_web_client.js:189
nimiq_web_client_bg.wasm:0x3e4cc4 Uncaught (in promise) RuntimeError: unreachable
    at nimiq_web_client_bg.wasm:0x3e4cc4
    at nimiq_web_client_bg.wasm:0x3f33db
    at nimiq_web_client_bg.wasm:0x404b00
    at nimiq_web_client_bg.wasm:0x403304
    at nimiq_web_client_bg.wasm:0x40379d
    at nimiq_web_client_bg.wasm:0x3f5eb8
    at nimiq_web_client_bg.wasm:0x326b3c
    at nimiq_web_client_bg.wasm:0x40600d
    at __wbg_adapter_40 (nimiq_web_client.js:220:10)
    at real (nimiq_web_client.js:189:20)
```

**Comment by viquezclaudio** [comment]
It was observed that sometimes the same peer is added, twice, too close to each other, to history sync:

```
Feb 23 21:12:20 light2 nimiq-client[58222]: 2023-02-23T21:12:20.189175147Z DEBUG syncer               | Adding peer PeerId("12D3KooWRNRgyKwnNeApYdyadsqDf6dMRNVT7gtYdVHtRvaMcbm4") into history sync
Feb 23 21:12:20 light2 nimiq-client[58222]: 2023-02-23T21:12:20.189210805Z INFO  sync                 | Requesting zkp from peer: PeerId("12D3KooWRNRgyKwnNeApYdyadsqDf6dMRNVT7gtYdVHtRvaMcbm4")
Feb 23 21:12:20 light2 nimiq-client[58222]: 2023-02-23T21:12:20.189241901Z DEBUG block_queue          | Received block via gossipsub block=#194401:MI:da5ddc07f8
Feb 23 21:12:20 light2 nimiq-client[58222]: 2023-02-23T21:12:20.189258394Z WARN  block_queue          | Discarding block #194401:MI:da5ddc07f8 outside of buffer window (max 194400)
Feb 23 21:12:20 light2 nimiq-client[58222]: 2023-02-23T21:12:20.189269143Z DEBUG syncer               | Adding peer PeerId("12D3KooWRNRgyKwnNeApYdyadsqDf6dMRNVT7gtYdVHtRvaMcbm4") into history sync
```
So, due to timing, we request macro headers to that peer twice, and then, so the second time we receive a response from that peer, we no longer have any outstanding requests from that peer. In that case, we just proceed to request epoch ids, to know if we are already synced with that peer or not

**Comment by jsdanielh** [comment]
Fixed in #1384.
